### PR TITLE
ECO-1519 mock data as json

### PR DESCRIPTION
**Pull request by liorama**
Mock offer data that used to be in CSV files are now JSON files.
Will change the data in https://github.com/kinecosystem/ecosystem-environments when this PR is approved.

### CODE REVIEW

File: data/offers/test.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
+const {performance, PerformanceObserver} = require('perf_hooks');
```
Comment (doodyparizada) [inline data/offers/test.js]: javascript?

File: scripts/src/admin/translations.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ import { ContentType, Offer, OfferContent } from "../models/offers";
 import { OfferTranslation } from "../models/translations";
 import { path } from "../utils/path";
 
-function parseContent(content: string): OfferContentContent {
+function  parseContent(content: string): OfferContentContent {
```
Comment (doodyparizada) [inline scripts/src/admin/translations.ts]: extra space

File: scripts/src/manage_db_data.ts [hunk 3]
```diff
@@ -113,98 +118,6 @@ async function parseSpend(data: string[][], appList: string[]) {
...
-		} else {
-			console.log(`poll type unknown: ${ v.get("PollPageType") }`);
-		}
-	}
-	if (offer) {
-		results.push(await createEarnInner(offer, poll));
-	}
-	return results.filter(v => !!v);
-}
-
```
Comment (doodyparizada) [inline scripts/src/manage_db_data.ts]: Yay!

File: scripts/src/manage_db_data.ts [hunk 4]
```diff
@@ -328,36 +241,63 @@ export async function initDb(scriptConfig: ScriptConfig, closeConnectionWhenDone
...
-				createOfferOptions.verbose && console.log(`created spend:${ contentType } offers`);
-			} else if (title === "Earn") {
-				results = await
-					parseEarn(parsed, contentType, appList, createOfferOptions);
-				createOfferOptions.verbose && console.log(`created earn:${ contentType } offers`);
-			} else {
-				throw new Error("Failed to parse " + parsed[0][0]);
+		const offerList: OfferData[] = require(path(join(offersDir, OFFERS_JSON_FILENAME)));
+		const offerContents: OfferContentData[] = require(path(join(offersDir, OFFERS_CONTENTS_JSON_FILENAME)));
+		const offerContentsDict = offerContents.reduce((acc: any, value) => {
```
Comment (doodyparizada) [inline scripts/src/manage_db_data.ts]: types

File: scripts/src/manage_db_data.ts [hunk 5]
```diff
@@ -328,36 +241,63 @@ export async function initDb(scriptConfig: ScriptConfig, closeConnectionWhenDone
...
+					KIN_BRAND,
+					offerMeta.title,
+					offerMeta.description,
+					offerMeta.image,
+					offer.amount,
+					DEFAULT_TOTAL_CAP,
+					DEFAULT_USER_CAP,
+					offerMeta.order_meta.title,
+					offerMeta.order_meta.description,
+					offer.content_type, JSON.parse(offer.content), appList, createOfferOptions);
```
Comment (doodyparizada) [inline scripts/src/manage_db_data.ts]: I think `createEarn` expects the unparsed offer.content.

File: scripts/src/manage_db_data.ts [hunk 6]
```diff
@@ -328,36 +241,63 @@ export async function initDb(scriptConfig: ScriptConfig, closeConnectionWhenDone
...
-			} else {
-				throw new Error("Failed to parse " + parsed[0][0]);
+		const offerList: OfferData[] = require(path(join(offersDir, OFFERS_JSON_FILENAME)));
+		const offerContents: OfferContentData[] = require(path(join(offersDir, OFFERS_CONTENTS_JSON_FILENAME)));
+		const offerContentsDict = offerContents.reduce((acc: any, value) => {
+			acc[value.offer_id] = value;
+			return acc;
+		}, {});
+		const offerData: Array<OfferData & OfferContentData> = offerList.map(value => Object.assign(value, offerContentsDict[value.id]));
+		await Promise.all(offerData.map(async offer => {
```
Comment (doodyparizada) [inline scripts/src/manage_db_data.ts]: Don't you think a simple for of loop is more readable? We don't need efficiency here

File: data/offers/test.js (hunk 1 above)
Comment (doodyparizada) [reply data/offers/test.js]: What is this even?

**Review by doodyparizada — CHANGES_REQUESTED** [review]

File: data/offers/test.js (hunk 1 above)
Comment (liorama) [reply data/offers/test.js]: I perfed what is the best way to merge two lists of objects, accidentally added the file.

File: scripts/src/manage_db_data.ts (hunk 5 above)
Comment (liorama) [reply scripts/src/manage_db_data.ts]: You are right, but for some reason PSQL exports the string quoted twice, so you have to json.parse it twice to get the JSON. I want to leave the exported files untouched so I'm just pre-parsing it. I'll add this explanation as a comment.

File: scripts/src/manage_db_data.ts (hunk 6 above)
Comment (liorama) [reply scripts/src/manage_db_data.ts]: even better, a for...loop just for you

File: scripts/src/create_data/offers.ts [hunk 7]
```diff
@@ -105,13 +105,13 @@ export async function createEarn(
...
 	orderTitle: string, orderDescription: string, contentType: ContentType,
-	offerContent: string, // of Quiz | Poll | Tutorial
+	offerContent: string, // string of Quiz | Poll | Tutorial
 	appList: string[] = [],
 	options: EarnOptions = {}): Promise<Offer | null> {
-
 	const existingOffer = await Offer.findOne({ name: offerName });
 	let offer;
 	let content;
+	console.log("creating earn offer %s of type %s associated apps: %s: ", offerName, contentType, appList);
```
Comment (doodyparizada) [inline scripts/src/create_data/offers.ts]: remove extra `': '` after last `%s`

File: scripts/src/mock_client.ts [hunk 8]
```diff
@@ -1860,7 +1860,7 @@ async function main() {
 		await walletSharedAcrossApps();
 	}
 
-	await v1();
+	// await v1();
```
Comment (doodyparizada) [inline scripts/src/mock_client.ts]: waaaaaat?

Comment (liorama) [reply scripts/src/mock_client.ts]: Screw v1... Who needs it anyway

**Pull request merged**